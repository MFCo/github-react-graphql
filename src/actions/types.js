import keyMirror from 'key-mirror';

export default keyMirror({
  UPDATE_USER: null,
  UPDATE_REPOSITORIES: null,
  ADD_REPOSITORIES: null,
  UPDATE_DIRECTORY: null,
  UPDATE_REPOSITORY: null
});