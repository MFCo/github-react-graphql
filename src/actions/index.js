import ACTIONS from './types';
import { createActions } from 'redux-actions';

export default createActions(...Object.keys(ACTIONS));