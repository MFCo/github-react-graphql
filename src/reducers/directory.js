import { handleActions } from 'redux-actions';

const defaultState = {
  list: [],
  id: null,
  stackTrace: []
};

export default handleActions({
  UPDATE_DIRECTORY: (state, { payload: { newList, newId } }) => {
    return {
      ...state,
      list: newList.sort((a, b) => a.type !== 'tree'),
      id: newId,
    }
  },
  FORWARD_STACK: (state, { payload: { newStack } }) => {
    return {
      ...state,
      stackTrace: state.stackTrace.concat(newStack)
    }
  },
  BACKWARD_STACK: (state) => {
    return {
      ...state,
      stackTrace: state.stackTrace.slice(0, state.stackTrace.length - 1)
    }
  }
}, defaultState);