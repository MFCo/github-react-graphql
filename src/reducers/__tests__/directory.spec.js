import directory from '../directory';
import actions from '../../actions';

it('should return the initial state', () => {
  expect(directory(undefined, {})).toMatchSnapshot()
})

it('should handle UPDATE_DIRECTORY', () => {
  expect(directory(undefined, actions.updateDirectory({
    newList:
      [
        {
          "name": ".babelrc",
          "type": "blob",
          "oid": "d42ea8be8b19750c64dc2a305cbd198739e9adbe"
        },
        {
          "name": ".dockerignore",
          "type": "blob",
          "oid": "c8f50f7cd317669d25aedba7415f8a1cbaadbab0"
        },
        {
          "name": ".eslintrc.js",
          "type": "blob",
          "oid": "29e7593c8ba8c3aedfcf2f78750e43ec428e367f"
        }
      ],
    newId: 'c8f50f7cd317669d25aedba7415f8a1cbaadbab0'
  }))).toMatchSnapshot()
})