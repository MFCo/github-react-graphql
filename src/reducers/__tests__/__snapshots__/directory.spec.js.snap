// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle BACKWARD_STACK 1`] = `
Object {
  "id": null,
  "list": Array [],
  "stackTrace": Array [],
}
`;

exports[`should handle FORWARD_STACK 1`] = `
Object {
  "id": null,
  "list": Array [],
  "stackTrace": Array [
    "c8f50f7cd317669d25aedba7415f8a1cbaadbab0",
  ],
}
`;

exports[`should handle UPDATE_DIRECTORY 1`] = `
Object {
  "id": "c8f50f7cd317669d25aedba7415f8a1cbaadbab0",
  "list": Array [
    Object {
      "name": ".eslintrc.js",
      "oid": "29e7593c8ba8c3aedfcf2f78750e43ec428e367f",
      "type": "blob",
    },
    Object {
      "name": ".dockerignore",
      "oid": "c8f50f7cd317669d25aedba7415f8a1cbaadbab0",
      "type": "blob",
    },
    Object {
      "name": ".babelrc",
      "oid": "d42ea8be8b19750c64dc2a305cbd198739e9adbe",
      "type": "blob",
    },
  ],
  "stackTrace": Array [],
}
`;

exports[`should return the initial state 1`] = `
Object {
  "id": null,
  "list": Array [],
  "stackTrace": Array [],
}
`;
